tests/attacks/evasion/test_lowprofool.py

```diff
@@ -0,0 +1,585 @@
+    labels_cancer = pd.Series(data=cancer['target'])
+
+    design_matrix_cancer_scaled, scaler_cancer = standardize(design_matrix_cancer)
+    clip_values = (design_matrix_cancer_scaled.min(), design_matrix_cancer_scaled.max())
+
+    return get_train_and_valid(design_matrix_cancer_scaled, labels_cancer), clip_values
+
+
+class SklearnLogisticRegression(BaseEstimator, LossGradientsMixin, ClassifierMixin):
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self._nb_classes = self.model.classes_.shape[0]
+
+    def fit(self, x: np.ndarray, y: np.array, **kwargs):
+        # Error-silencer only. Not used here.
+        raise NotImplementedError(
+            "Method `fit` of SklearnLogisticRegression class is a placeholder.")
+
+    @property
+    def input_shape(self):
+        """
+        Return the shape of one input sample.
+
+        :return: Shape of one input sample.
+        """
+        if len(self.model.coef_.shape) == 1:
+            return self.model.coef_.shape[0],
+        else:
+            return self.model.coef_.shape[1],
+
+    def loss_gradient(self, x: np.ndarray, y: np.ndarray, **kwargs) -> np.ndarray:
+        """
+        Compute the gradient of the loss function w.r.t. `x`.
+
+        :param x: Sample input with shape as expected by the model.
+        :param y: Target values (class labels) one-hot-encoded of shape `(nb_samples, nb_classes)`.
+        :param kwargs:
+        :return: Array of gradients of the same shape as `x`.
+        """
+        x_probas = self.model.predict_proba(x)
+        errors = x_probas - y
+        thetas = self.model.coef_
+
+        if thetas.shape[0] == 1:
+            thetas = np.append(thetas * (-1), thetas, axis=0)
+
+        return (errors @ thetas) / self.model.classes_.size
+
+    def predict(self, x: np.ndarray, **kwargs) -> np.ndarray:
+        """
+        Perform prediction of the estimator for input `x`.
+
+        :param x: Samples.
+        :param kwargs:
+        :return: Predictions by the model.
+        """
+        return self.model.predict_proba(x)
+
+
+class NeuralNetwork:
+    def __init__(self):
+        self.loss_fn = torch.nn.MSELoss(reduction='sum')
+
+    @staticmethod
+    def get_nn_model(input_dimensions, output_dimensions, hidden_neurons):
+        return torch.nn.Sequential(
+            nn.Linear(input_dimensions, hidden_neurons),
+            nn.ReLU(),
+            nn.Linear(hidden_neurons, output_dimensions),
+            nn.Softmax(dim=1)
+        )
+
+    def train_nn(self, nn_model, x, y, learning_rate, epochs):
+        for _ in range(epochs):
+            y_pred = nn_model.forward(x)
+            loss = self.loss_fn(y_pred, y)
+            nn_model.zero_grad()
+            loss.backward()
+            with torch.no_grad():
+                for param in nn_model.parameters():
+                    param.data -= learning_rate * param.grad
+
+
+class TestLowProFool(unittest.TestCase):
```

Comment:
Please convert to pytest format similar to the other tests in `tests/attacks/evasion`.